src/components/EquipmentList.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { FunctionComponent } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import EquipmentExpansionList from "./EquipmentExpansionList";
+import Equipment from "../resources/Equipment";
+import EquipmentStandardList from "./EquipmentStandardList";
+
+const useStyles = makeStyles({
+  root: {
+    width: "100%",
+  },
+});
+
+interface EquipmentListProps {
+  equipmentList: Equipment[] | undefined;
+  currentCategory: string;
+  setFieldValue: (field: string, value: number | string | boolean) => void;
+  quantities: {
+    [k: string]: number;
+  };
+}
+const EquipmentList: FunctionComponent<EquipmentListProps> = ({
+  equipmentList,
+  currentCategory,
+  quantities,
+  setFieldValue,
+}) => {
+  const classes = useStyles();
+  if (equipmentList) {
+    const category = [...new Set(equipmentList.map((items) => items.category))];
```

Comment:
I definitely approve this syntax.
I do wonder if this loop through the equipment list could be stored somewhere so we don't keep doing this on each render.
Also wondering what the complete list of loop processing is.
Worth considering is just getting the categories from the database.
```suggestion
    const category = [...new Set(equipmentList.map((item) => item.category))];
```

Reply:
resolved in 58c2e41